app/backend/src/clj/teet/migration/activity_procurement_info.clj

```diff
@@ -21,3 +21,15 @@
                           {:activity/procurement-id procurement-id})
                         (when procurement-nr
                           {:activity/procurement-nr procurement-nr})))})))
+
+(defn retract-procurement-info-from-activity [conn]
+  (let [activities (d/q '[:find ?a
+                          :where [?a :activity/procurement-id _]
+                          [?a :activity/procurement-nr _]]
+                     (d/db conn))
+        procurement-ids (for [[activity-id] activities]
+                          [:db/retract activity-id :activity/procurement-id])
+        procurement-nrs (for [[activity-id] activities]
+          [:db/retract activity-id :activity/procurement-nr])]
```

Comment:
`(let [activities (->> (d/q '[:find ?a
                              :where [?a :activity/procurement-id _]
                              [?a :activity/procurement-nr _]]
                            (d/db conn))
                       (mapv first))
        procurement-ids (for [activity-id activities]
                          [:db/retract activity-id :activity/procurement-id])
        procurement-nrs (for [activity-id activities]
                          [:db/retract activity-id :activity/procurement-nr])]`

I think it's a bit more clear if you mapv first out of the activities list and then you don't need the destructuring of the list elements in the for loop.